### PR TITLE
Add tf2onnx convert function with unittest, and convert CLI

### PR DESCRIPTION
**Pull request by STALLAAAA**
Closes #

1.add tensorflow to onnx convert function and unittest
2.add convert cli
3.add requirement for tf2onnx 
4.add function for datatype transferring--Datatype to tf

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: tests/test_tensorflow_conversion.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        onnx_model = convert(self.mobilenet_save_path, 'tensorflow', 'onnx')
+        onnx.checker.check_model(onnx_model)
+        sess = onnxruntime.InferenceSession(onnx_model.SerializeToString())
+        input_name = sess.get_inputs()[0].name
+        output_name = sess.get_outputs()[0].name
+        res = np.array(sess.run([output_name], {input_name: self.x}))
+        onnxres = self.imagenet_labels[np.argsort(res)[0, ::-1][:5] + 1]
+        pred = onnxres[-1][-5:]
+        np.testing.assert_string_equal(str(pred[::-1]), str(self.decoded))
+    if __name__ == '__main__':
```
Comment (YuanmingLeee) [inline tests/test_tensorflow_conversion.py:48]: remove this

File: modelci/cli/modelhub.py [hunk 2]
```diff
@@ -242,3 +244,194 @@ def delete(model_id: str = typer.Argument(..., help='Model ID')):
...
+        'IMAGE_CLASSIFICATION': saveTask.IMAGE_CLASSIFICATION,
+        'SEGMENTATION': saveTask.SEGMENTATION,
+        'OBJECT_DETECTION': saveTask.OBJECT_DETECTION
+    }
+    in_format = {
+            'FORMAT_NONE': ModelInputFormat.FORMAT_NONE,
+            'FORMAT_NHWC': ModelInputFormat.FORMAT_NHWC,
+            'FORMAT_NCHW': ModelInputFormat.FORMAT_NCHW,
+    }
+    if way == ('keras', 'onnx'):
```
Comment (YuanmingLeee) [inline modelci/cli/modelhub.py:288]: add a todo, to extract the common parts for building the conversion parameters

**Review by YuanmingLeee — COMMENTED** [review]
Other parts look fine

File: modelci/cli/modelhub.py [hunk 3]
```diff
@@ -242,3 +244,194 @@ def delete(model_id: str = typer.Argument(..., help='Model ID')):
...
+        import tensorflow as tf
+        save_path = generate_path(model_name=model_name,
+                                  framework=saveFramework.TENSORFLOW,
+                                  task=task[model_task],
+                                  engine=saveEngine.TFS,
+                                  version=version)
+        save_path.parent.mkdir(parents=True, exist_ok=True)
+        loaded = tf.saved_model.load(model_path)
+        cvt.convert(model=loaded, src_framework='tensorflow', dst_framework='tfs', save_path=save_path)
+    if way == ('xgboost', 'onnx'):
```
Comment (huangyz0918) [inline modelci/cli/modelhub.py:322]: If I were you, I would write something like 

```python
if way[0] == 'xgboost':
    import onnx
    import torch
    save_engine = saveEngine.PYTORCH
    save_framework = saveFramework.PYTORCH
    if is_onnx:
        save_framework = saveFramework.ONNX
        save_engine = saveEngine.ONNX
        
    loaded = torch.load(model_path)
    inputs = [convertIO(shape=eval(shape_input), dtype=eval(dtype_input), name='input_0')]
    onnx_model = cvt.convert(model=loaded, src_framework='xgboost', dst_framework='onnx', inputs=inputs)
    onnx.checker.check_model(onnx_model)
    save_path = generate_path(model_name=model_name, framework=save_framework,
                            task=task[model_task], engine=save_engine, version=version)
    save_path.parent.mkdir(parents=True, exist_ok=True)

    if is_onnx:
        onnx.save(onnx_model, save_path)
    else:
        torch.save(torch_model, save_path)
```

Try to merge some if-elses together to reduce the code redundancy.

File: tests/test_tensorflow_conversion.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    def setUpClass(cls):
+        tmpdir = tempfile.mkdtemp()
+
+        file = tf.keras.utils.get_file(
+            "grace_hopper.jpg",
+            "https://storage.googleapis.com/download.tensorflow.org/example_images/grace_hopper.jpg")
+        img = tf.keras.preprocessing.image.load_img(file, target_size=[224, 224])
+        cls.x = tf.keras.preprocessing.image.img_to_array(img)
+        cls.x = tf.keras.applications.mobilenet.preprocess_input(
+            cls.x[tf.newaxis, ...])
```
Comment (huangyz0918) [inline tests/test_tensorflow_conversion.py:25]: remove line wrap here.

File: modelci/cli/modelhub.py [hunk 5]
```diff
@@ -242,3 +244,194 @@ def delete(model_id: str = typer.Argument(..., help='Model ID')):
...
+
+        model_task: Optional[str] = typer.Option(..., '-t', '--task', help='task of model'
+                                                                           'such as \'IMAGE_CLASSIFICATION\''
+                                                                           'or \'SEGMENTATION\'\'OBJECT_DETECTION\'')
+):
+    """
+    TODO replace input and output str with IOShape
+    """
+    import modelci.hub.converter.converter as cvt
+    way = (src, dst)
```
Comment (huangyz0918) [inline modelci/cli/modelhub.py:277]: `way` is an ambiguous word.

File: modelci/cli/modelhub.py (hunk 2 above)
Comment (STALLAAAA) [reply modelci/cli/modelhub.py:288]: I am working on this restruction right now. I will make this change in next commit

File: modelci/cli/modelhub.py [hunk 6]
```diff
@@ -242,3 +244,194 @@ def delete(model_id: str = typer.Argument(..., help='Model ID')):
...
+        dtype_output: Optional[str] = typer.Option(None, '-do', '--dtypeo', help='output data type such as \'float\''),
+
+        model_task: Optional[str] = typer.Option(..., '-t', '--task', help='task of model'
+                                                                           'such as \'IMAGE_CLASSIFICATION\''
+                                                                           'or \'SEGMENTATION\'\'OBJECT_DETECTION\'')
+):
+    """
+    TODO replace input and output str with IOShape
+    """
+    import modelci.hub.converter.converter as cvt
```
Comment (huangyz0918) [inline modelci/cli/modelhub.py:276]: converter should be converter, I'm quite not understand what is `cvt` means, no one will blame you to add 7 extra  words.

File: modelci/cli/modelhub.py [hunk 7]
```diff
@@ -26,7 +26,9 @@
 from modelci.ui import model_view, model_detailed_view
 from modelci.utils import Logger
 from modelci.utils.misc import remove_dict_null
-
+from modelci.types.bo import Framework as saveFramework, Engine as saveEngine, Task as saveTask, IOShape as convertIO
```
Comment (YuanmingLeee) [inline modelci/cli/modelhub.py:29]: Why you change the import name? I think Framework, Engine, ... are fine

File: modelci/cli/modelhub.py (hunk 3 above)
Comment (STALLAAAA) [reply modelci/cli/modelhub.py:322]: Thank you very much, I am working on this right now

File: modelci/cli/modelhub.py (hunk 7 above)
Comment (STALLAAAA) [reply modelci/cli/modelhub.py:29]: ''Framework'', ''Engine'',.. are already used by 

>24  from modelci.types.models import Framework, Engine, IOShape, Task, Metric, ModelUpdateSchema

Currently all convert functions are only compatible with  'saveFramework'...

Comment (YuanmingLeee) [reply modelci/cli/modelhub.py:29]: my advice is to change them to `FrameworkLagacy`, ... to indicate that we gonna remove them in future

Comment (YuanmingLeee) [reply modelci/cli/modelhub.py:29]: `saveFramework` is not clear

File: tests/test_tensorflow_conversion.py (hunk 1 above)
Comment (STALLAAAA) [reply tests/test_tensorflow_conversion.py:45]: Do you mean deleting the entire unit test for tensorflow to onnx?
